src/journal/journald-server.c

```diff
@@ -1378,7 +1378,7 @@ int server_process_datagram(sd_event_source *es, int fd, uint32_t revents, void
 
         if (fd == s->syslog_fd) {
                 if (n > 0 && n_fds == 0)
-                        server_process_syslog_message(s, strstrip(s->buffer), ucred, tv, label, label_len);
+                        server_process_syslog_message(s, s->buffer + strspn(s->buffer, WHITESPACE), ucred, tv, label, label_len);
```

Comment:
Hm, unrelated to this PR, but we use `IOVEC_SET_STRING(iovec, buffer)` and drop all bytes after '\0':
```sh
$ perl -e 'print "hey\0ho"' | logger -t HM
$ journalctl -t HM -o verbose
...
    MESSAGE=hey
...
```